pkg/validation/validation.go

```diff
@@ -0,0 +1,51 @@
+package validation
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const ErrorNotFoundAgent = "unsupported fence agent %s:"
+
+type validateAgentExistence struct {
+	isFileExist func(string) (fs.FileInfo, error)
```

Comment:
- for better readabilty, what about introducing a funtion type instead of using the finction ignatire at multiple places?
- FileInfo is never used (which good, because:)
- name should be independent of the implementation, so maybe don't use anything about files at all?